### ISSUE TITLE
False positive for `Lint/ReturnInVoidContext`

### ISSUE THREAD

**Issue by dvandersluis** [rubocop/rubocop#9675]
```ruby
def foo=(x)
  return super unless x.is_a?(String)
  self[:foo] = x
end
```

This registers `Lint/ReturnInVoidContext` on `return super...` but I think this is a valid use case.

--------

## RuboCop version

```
$ [bundle exec] rubocop -V
1.12.1 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-darwin18)
  - rubocop-performance 1.10.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.2.0
```

**Comment by dvandersluis** [comment]
@rubocop/rubocop-core do you guys agree this is a false positive? If so I'll fix it.

**Comment by koic** [comment]
Interesting! The return value of `super` for `def foo=(x)` should also be void. Therefore `return super` to void method may not be preferable.
For example, dynamic typed Ruby works, but static typed Java occurs a compile error.

So, I think it's a false positive as a type semantics, but I'd like to hear other opinions.

## Example for no return void

```java
% cat Hello.java
public class Hello extends Base {
    public static void main(String[] args) {
       new Hello().say("hi");
    }

    public void say(String message) {
        super.say(message);
        System.out.println("[sub]" + message);
    }

    class Base {
        public void say(String message) {
            System.out.println("[super]" + message);
        }
    }
}

% javac Hello.java

% java Hello
[super]hi
[sub]hi
```

## Example for return void

diff for the above example code.

```diff
% git diff Hello.java
diff --git a/Hello.java b/Hello.java
index 1422081..23fb2e9 100644
--- a/Hello.java
+++ b/Hello.java
@@ -4,7 +4,7 @@ public class Hello extends Base {
     }

     public void say(String message) {
-        super.say(message);
+        return super.say(message);
         System.out.println("[sub]" + message);
     }

```

compile error.

```java
% cat Hello.java
public class Hello extends Base {
    public static void main(String[] args) {
       new Hello().say("hi");
    }

    public void say(String message) {
        return super.say(message);
        System.out.println("[sub]" + message);
    }

    class Base {
        public void say(String message) {
            System.out.println("[super]" + message);
        }
    }
}

% javac Hello.java
Hello.java:7: error: incompatible types: unexpected return value
        return super.say(message);
                        ^
1 error
```
